CMakeLists.txt

```diff
@@ -35,7 +35,9 @@ if(NOT CYGWIN)
   set_property(TARGET cquery PROPERTY CXX_EXTENSIONS OFF)
 endif()
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
+if(${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC OR 
+   (${CMAKE_SYSTEM_NAME} STREQUAL Windows AND 
+    ${CMAKE_CXX_COMPILER_ID} STREQUAL Clang))
```

Comment:
Why not use `if(MSVC)`? Won't that work with clang-cl?

Because this check won't work with Clang on MSYS2 (not clang-cl).

Reply:
Does invoking cmake with Clang as the compiler in msys2 result in CMAKE_SYSTEM_NAME being Windows and CMAKE_CXX_COMPILER_ID being Clang? CMake docs only say that MSVC is set when the compiler is Visual C++. I'm assuming it won't be set if the compiler is Clang

Reply:
I know that trying to use regular Clang (not clang-cl) on Windows is not supported by CMake and results in an error. I'm not sure if the situation changes with msys2